### PR TITLE
Update the DatabaseHelper after the March 2021 patch.

### PR DESCRIPTION
**Pull request by ChrisPJohn**
I modified the GenDatabaseHelper.py script to not generate casts since as far as I could tell they were only duplicate lines based on the asset file I was working with.

### CODE REVIEW

File: Scripts/GenDatabaseHelper.py [hunk 1]
```diff
@@ -61,9 +61,8 @@ def write_subclass(file, name, asset_list, indent = 0):
 def write_getter(file, asset, indent = 0):
     member_name = asset.member_name()
     cast = ""
-    if asset.database_type != asset.asset_type:
-        cast = f"({asset.asset_type})"
-    write_line(file, f"public static {asset.asset_type} {member_name} => {cast}DatabaseRepository.GetDatabase<{asset.database_type}>().TryGetElement(\"{asset.name}\", \"{asset.guid}\");", indent)
+    if asset.database_type == asset.asset_type:
+        write_line(file, f"public static {asset.asset_type} {member_name} => {cast}DatabaseRepository.GetDatabase<{asset.database_type}>().TryGetElement(\"{asset.name}\", \"{asset.guid}\");", indent)
```
Comment (spacehamster) [inline Scripts/GenDatabaseHelper.py:65]:
```suggestion
        write_line(file, f"public static {asset.asset_type} {member_name} => DatabaseRepository.GetDatabase<{asset.database_type}>().TryGetElement(\"{asset.name}\", \"{asset.guid}\");", indent)
```

File: Scripts/GenDatabaseHelper.py [hunk 2]
```diff
@@ -61,9 +61,8 @@ def write_subclass(file, name, asset_list, indent = 0):
 def write_getter(file, asset, indent = 0):
     member_name = asset.member_name()
     cast = ""
```
Comment (spacehamster) [inline Scripts/GenDatabaseHelper.py:63]:
```suggestion
```

**Review by spacehamster — CHANGES_REQUESTED** [review]
The database format has changed. Previously there were no duplicate entries and a subclass entry could be contained in a superclass database with no matching subclass database, which no longer holds. Using `AbilityCheckAffinityBeltOfDwarvenkind` as an example.
In 0.4.20
```
Name                                    Type                                    DatabaseType
AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinition
```
In 0.5.41
```
Name                                    Type                                    DatabaseType
AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinition
AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinitionAffinity
AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinitionAbilityCheckAffinity
```
The new code assumes that there is always an entry where the entry Type and DatabaseType match, so the cast is no longer necessary and should be removed.

**Comment by ChrisPJohn** [comment]
> The database format has changed. Previously there were no duplicate entries and a subclass entry could be contained in a superclass database with no matching subclass database, which no longer holds. Using `AbilityCheckAffinityBeltOfDwarvenkind` as an example.
> In 0.4.20
> 
> ```
> Name                                    Type                                    DatabaseType
> AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinition
> ```
> 
> In 0.5.41
> 
> ```
> Name                                    Type                                    DatabaseType
> AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinition
> AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinitionAffinity
> AbilityCheckAffinityBeltOfDwarvenkind   FeatureDefinitionAbilityCheckAffinity   FeatureDefinitionAbilityCheckAffinity
> ```
> 
> The new code assumes that there is always an entry where the entry Type and DatabaseType match, so the cast is no longer necessary and should be removed.

Good catch. I had stopped using the cast but forgot to go back and fully remove it once I verified it was working.

**Pull request merged**